### DISCUSSION TITLE
How to prevent re-render

### DISCUSSION THREAD

**Discussion by cmpscabral** [mantinedev/mantine#2684] [Q&A] [answered]
I'm sure this is a basic React question, please bear with me.

I have a simple list of images and every time I open/close a modal, this list gets re-rendered (and fetched from cache/server).
Is there a way to prevent this?

Simple example: https://codesandbox.io/s/vigilant-swanson-4w2rkj

**Comment by rtivital** [comment, marked as answer]
https://codesandbox.io/s/ecstatic-tamas-8zxmd4?file=/src/App.js

**Comment by cmpscabral** [comment]
Thank you!
